### PR TITLE
Add self-hosted runners for FreeBSD and Ubuntu arm64

### PR DESCRIPTION
**Pull request by quarckster**
##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -107,6 +107,25 @@ jobs:
...
+  self-hosted-runners:
+    strategy:
+      matrix:
+        os: [freebsd-13.2, ubuntu-arm64-22.04]
+    runs-on: ${{ matrix.os }}-self-hosted
+    continue-on-error: true
+    steps:
+    - uses: actions/checkout@v4
+    - name: config
+      run: ./config
```
Comment (t8m) [inline .github/workflows/ci.yml:119]: I'd suggest enabling some more stuff here - i.e., add `enable-fips enable-ec_nistp_64_gcc_128 enable-md2 enable-rc5 enable-ssl3 enable-ssl3-method enable-trace`

**Review by t8m — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -107,6 +107,25 @@ jobs:
     - name: make test
       run: make test HARNESS_JOBS=${HARNESS_JOBS:-4}
 
+  self-hosted-runners:
```
Comment (t8m) [inline .github/workflows/ci.yml:110]: Make the name shorter (`our-runners` or `self-hosted`) so the os from the matrix is not truncated in the GitHub UI.

**Review by t8m — APPROVED** [review]

**Review by arapov — APPROVED** [review]

**Comment by t8m** [comment]
Merged to the master, 3.2, 3.1 and 3.0 branches. Thank you.

File: .github/workflows/os-zoo.yml [hunk 3]
```diff
@@ -166,3 +166,22 @@ jobs:
     - name: test
       working-directory: _build
       run: nmake test VERBOSE_FAILURE=yes HARNESS_JOBS=4
+  
+  self-hosted:
```
Comment (levitte) [inline .github/workflows/os-zoo.yml:170]: All other jobs are per platform...  it's a little strange that this breaks that pattern.

Comment (quarckster) [reply .github/workflows/os-zoo.yml:170]: Otherwise, we would have duplicated code.

Comment (levitte) [reply .github/workflows/os-zoo.yml:170]: I get that, and boy do we have that already!

It's just a little strange to see this lots of lines saying "alpine" or "linux" or "macos" or "windows"...  and then "self-hosted".

Comment (t8m) [reply .github/workflows/os-zoo.yml:170]: IMO it is fine as is.

**Pull request closed without merging**